lib/cas_application_renewal.rb

```diff
@@ -0,0 +1,16 @@
+class CasApplicationRenewal
+  def initialize
+  end
+
+  RENEWAL_AFTER = 1.year
+
+  def renewals
+    potential_renewals = Project.cas_access_granted
+    potential_renewals.each do |project|
```

Comment:
I think there's a way to this via direct db query.
e.g every time this is run we might be checking hundreds in an enumerable.

will discuss on call